internal/services/recoveryservices/recovery_services_vault_data_source.go

```diff
@@ -4,78 +4,113 @@
+	}
+}
+func (SiteRecoveryRecoveryVaultDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.LocationComputed(),
 
-			"location": commonschema.LocationComputed(),
+		"tags": commonschema.TagsDataSource(),
 
-			"tags": commonschema.TagsDataSource(),
+		"identity": commonschema.SystemAssignedUserAssignedIdentityComputed(),
 
-			"sku": {
-				Type:     pluginsdk.TypeString,
-				Computed: true,
-			},
+		"sku": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
 		},
 	}
 }
 
-func dataSourceRecoveryServicesVaultRead(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).RecoveryServices.VaultsClient
-	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
-	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
-	defer cancel()
+func (r SiteRecoveryRecoveryVaultDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.RecoveryServices.VaultsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
 
-	id := vaults.NewVaultID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
-	resp, err := client.Get(ctx, id)
-	if err != nil {
-		if response.WasNotFound(resp.HttpResponse) {
-			return fmt.Errorf("%s was not found", id)
-		}
+			var recoveryServiceVault SiteRecoveryRecoveryVaultDataSourceModel
+			if err := metadata.Decode(&recoveryServiceVault); err != nil {
+				return err
+			}
 
-		return fmt.Errorf("retrieving %s: %+v", id, err)
-	}
+			id := vaults.NewVaultID(subscriptionId, recoveryServiceVault.RecoveryGroupName, recoveryServiceVault.Name)
+			resp, err := client.Get(ctx, id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("recovery service vault %s not found", id.ID())
+				}
+				return fmt.Errorf("retreiving %s: %+v", id, err)
+			}
 
-	if resp.Model == nil {
-		return fmt.Errorf("retrieving %s: `model` was nil", id)
-	}
-	model := resp.Model
+			model := resp.Model
 
-	d.SetId(id.ID())
-	d.Set("name", id.VaultName)
-	d.Set("resource_group_name", id.ResourceGroupName)
-	d.Set("location", location.Normalize(model.Location))
+			flattenedIdentity, err := identity.FlattenSystemAndUserAssignedMapToModel(model.Identity)
 
-	skuName := ""
-	if model.Sku != nil {
-		skuName = string(model.Sku.Name)
-	}
-	d.Set("sku", skuName)
+			if err != nil {
+				return fmt.Errorf("flattening `identity`: %+v", err)
+			}
+
+			skuName := ""
+			if model.Sku != nil {
+				skuName = string(model.Sku.Name)
+			}
+
+			recoveryServiceVault.Sku = skuName
+			recoveryServiceVault.Location = location.Normalize(model.Location)
+			recoveryServiceVault.Tags = pointer.From(model.Tags)
+			recoveryServiceVault.Identity = pointer.From(flattenedIdentity)
```

Comment:
For consistency and to prevent any crashes can we refactor this to:

```
if model := resp.Model; model != nil {
    // do all the flattening and setting into state here
}
```